MapboxCoreNavigation/RouteController.swift

```diff
@@ -174,7 +174,10 @@ open class RouteController: NSObject, Router {
     @objc func didChangeApplicationState() {
         eventsManager.reportChange(to: UIApplication.shared.applicationState)
     }
-
+    @objc private func didRecieveTerminationWarning() {
+        endNavigation()
```

Comment:
Shouldn't this already be handled by deinit?

Reply:
@bsudekum Believe it or not, `deinit` is not called in a force-quit scenario. 

Demo:

```swift
import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.willTerminate), name: .UIApplicationWillTerminate, object: nil)
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @objc func willTerminate() {
        print("Will Terminate!")
    }
    
    deinit {
        print("Deinit called!")
        NotificationCenter.default.removeObserver(self)
    }
}
```

if you run this in a sample project, and force-quit the app by "swiping up" on the app in the app-chooser, you'll notice that the only thing printed out onto the console is `Will Terminate!`

Reply:
Yep. relying on `deinit()` or `-dealloc` can be problematic in a variety of scenarios.